### PR TITLE
Enhanced Video to allow moving along video timeline with keyboard

### PR DESCRIPTION
**Pull request by jcfilben**
#### What does this PR do?
Adds the ability to move forward and backwards in a video with the left and right arrow keys. The default amount of time to move forward/backward in the video is 5 seconds but this can be customized with the `skipInterval` prop.

#### What testing has been done on this PR?
Tested with the existing video storybook stories and customizing the time with the `skipInterval` prop.

#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `userEvent` is used in place of `fireEvent`.
- [ ] `asFragment()` is used for snapshot testing.

#### What are the relevant issues?
https://github.com/grommet/grommet/issues/6021

#### Do the grommet docs need to be updated?
Yes, I will add a PR for the grommet doc changes

#### Should this PR be mentioned in the release notes?
Yes

#### Is this change backwards compatible or is it a breaking change?
Compatible

### CODE REVIEW

File: src/js/components/Video/StyledVideo.js [hunk 1]
```diff
@@ -11,18 +11,18 @@ const FIT_MAP = {
...
   max-width: 100%;
   z-index: 1;
-  ${props => props.fit && fitStyle} ::cue {
-    background: ${props => props.theme.video.captions.background};
+  ${(props) => props.fit && fitStyle} ::cue {
+    background: ${(props) => props.theme.video.captions.background};
   }
 
-  ${props => props.theme.video && props.theme.video.extend};
+  ${(props) => props.theme.video && props.theme.video.extend};
```
Comment (britt6612) [inline src/js/components/Video/StyledVideo.js:25]: are these changes intended these linting ?

Comment (jcfilben) [reply src/js/components/Video/StyledVideo.js:25]: Those were just changes from `yarn lint`

File: src/js/components/Video/Video.js [hunk 2]
```diff
@@ -210,6 +212,16 @@ const Video = forwardRef(
       [duration, videoRef],
     );
 
+    const seekForward = useCallback(() => {
+      setInteracting(true);
+      videoRef.current.currentTime += skipInterval || 5;
```
Comment (britt6612) [inline src/js/components/Video/Video.js:217]: Is 5 just the standard number? Just want to know how you decided on 5

File: src/js/components/Video/StyledVideo.js [hunk 3]
```diff
@@ -35,6 +35,14 @@ const StyledVideoContainer = styled.div`
   overflow: hidden;
   position: relative;
   ${genericStyles};
+  &:focus {
```
Comment (britt6612) [inline src/js/components/Video/StyledVideo.js:38]: I like the idea of the line for the focus just thinking is someone didn't want this or wanted a horizontal line

File: src/js/components/Video/Video.js (hunk 2 above)
Comment (jcfilben) [reply src/js/components/Video/Video.js:217]: I saw other video players using 5 seconds but I couldn't find any info on if there is a standard for this

File: src/js/components/Video/StyledVideo.js [hunk 4]
```diff
@@ -35,6 +35,14 @@ const StyledVideoContainer = styled.div`
...
+    outline-offset: 0px;
+    outline: ${(props) => {
+      const color = normalizeColor('focus', props.theme);
+      if (color) return `2px solid ${color};`;
```
Comment (MikeKingdom) [inline src/js/components/Video/StyledVideo.js:42]: Should this come from utils/styles.js focusStyles?

File: src/js/components/Video/Video.js (hunk 2 above)
Comment (MikeKingdom) [reply src/js/components/Video/Video.js:217]: Netflix and I think YouTube tend to use 10s, at list on their skip forward and backward buttons.

Comment (halocline) [reply src/js/components/Video/Video.js:217]: Might be nice to have the default specified in the theme, so that callers can set their own defaults as opposed to each instance.

Comment (jcfilben) [reply src/js/components/Video/Video.js:217]: If we have the default set in the theme should we still have the `skipInterval` prop to override the default?

File: src/js/components/Video/StyledVideo.js (hunk 4 above)
Comment (jcfilben) [reply src/js/components/Video/StyledVideo.js:42]: Yes, I will go back and change that

File: src/js/components/Video/Video.js (hunk 2 above)
Comment (halocline) [reply src/js/components/Video/Video.js:217]: yes, I would keep the prop -- nice to have ability to override and not lock folks in

Comment (jcfilben) [reply src/js/components/Video/Video.js:217]: Sounds good, I will make these changes

File: src/js/themes/base.js [hunk 5]
```diff
@@ -1682,6 +1682,7 @@ export const generate = (baseSpacing = 24, scale = 6) => {
       // scrubber: { track: { color: undefined } },
       scrubber: {
         color: 'light-4',
+        interval: 10,
```
Comment (halocline) [inline src/js/themes/base.js:1685]: Needed in base.d.ts as well?

Comment (halocline) [inline src/js/themes/base.js:1685]: Commented out Line 1682 looks suspect. Can we address the `scrubber.track.color` properties? Consolidate under the active scrubber property + get 100% alignment here and  `base.d.js`.

File: src/js/components/Video/stories/typescript/ControlsBelow.tsx [hunk 6]
```diff
@@ -4,7 +4,7 @@ import { Box, Video } from 'grommet';
 
 const SimpleVideo = (props) => (
   <Box align="center" pad="large">
-    <Video {...props}>
+    <Video skipInterval={2} {...props}>
```
Comment (halocline) [inline src/js/components/Video/stories/typescript/ControlsBelow.tsx:7]: I am having troubles accessing the player controls after I play the video. They are available on first render, but if I play, they get hidden and I am no longer able to access them with the keyboard. Am I doing something wrong?

**Review by britt6612 — APPROVED** [review]
Looks good

File: src/js/components/Video/propTypes.js [hunk 7]
```diff
@@ -36,6 +36,7 @@ if (process.env.NODE_ENV !== 'production') {
       volumeDown: PropTypes.string,
       volumeUp: PropTypes.string,
     }),
+    skipInterval: PropTypes.number,
```
Comment (halocline) [inline src/js/components/Video/propTypes.js:39]: While we are in here, can we make sure that the propTypes and index.d.ts definitions are at parity and easy to compare?
- first level of properties, can we alphabetize?
- messages properties, can we alphabetize?
- `controls.items` is missing enum prop type

I think that's all.

File: src/js/components/Video/Video.js [hunk 8]
```diff
@@ -455,53 +469,56 @@ const Video = forwardRef(
...
-            setPercentagePlayed((video.currentTime / video.duration) * 100);
-            if (onTimeUpdate) onTimeUpdate(event);
-          }}
-          onVolumeChange={(event) => {
-            setVolume(videoRef.current.volume);
-            if (onVolumeChange) onVolumeChange(event);
-          }}
-          autoPlay={autoPlay || false}
-          loop={loop || false}
+      <Keyboard onRight={seekForward} onLeft={seekBackward}>
```
Comment (halocline) [inline src/js/components/Video/Video.js:472]: I noted an issue regarding the lack of ability to access the controls after video play. Would it be appropriate to add an `onTab` here with a callback that sets `setInteracting(true)`?

**Review by halocline — COMMENTED** [review]
Looking good. Noted a few items for tidying up.

File: src/js/components/Video/Video.js (hunk 8 above)
Comment (jcfilben) [reply src/js/components/Video/Video.js:472]: Yes I think that is a good idea. I will add that

File: src/js/components/Video/stories/typescript/ControlsBelow.tsx (hunk 6 above)
Comment (jcfilben) [reply src/js/components/Video/stories/typescript/ControlsBelow.tsx:7]: resolve in latest commit

**Review by halocline — APPROVED** [review]
Thanks for the cleanups and adding the focus / interacting events.

Outside of this PR's scope, the controls' focus gets cutoff -- do we have an issue for that?

**Comment by jcfilben** [comment]
> Outside of this PR's scope, the controls' focus gets cutoff -- do we have an issue for that?

Screen Shot 2022-04-05 at 9 29 56 AM

Just want to confirm that you are talking about how you can't see the focus indicator when interacting with keyboard on the controls menu. If so then no we do not have an issue for this but I can create one

**Comment by jcfilben** [comment]
Issue created: https://github.com/grommet/grommet/issues/6054

**Review by ericsoderberghp — APPROVED** [review]
